### ISSUE TITLE
initial cache-key resolution should display whole key, not just a short SHA

### ISSUE THREAD

**Issue by Cynical-Optimist** [buildstream-migration/bst-staging#343]
[See original issue on GitLab](https://gitlab.com/BuildStream/buildstream/-/issues/343)
In GitLab by [[Gitlab user @devcurmudgeon]](https://gitlab.com/devcurmudgeon) on Apr 5, 2018, 19:50

Current log is as below. The short shas can't used as a definitive list of what was being built. reviewing this log much later on big projects, we'd be at risk of finding clashes.

```
fetch needed 20b0640d gnu-toolchain/base.bst 
fetch needed b57571d4 gnu-toolchain/stage1-binutils.bst 
fetch needed 2d81f140 gnu-toolchain/stage1-gcc.bst 
     waiting 15871240 gnu-toolchain/stage1.bst 
fetch needed 1c5173b0 gnu-toolchain/stage2-fhs-dirs.bst 
fetch needed 238d8701 gnu-toolchain/stage2-linux-api-headers.bst 
fetch needed 78eddec5 gnu-toolchain/stage2-glibc.bst 
fetch needed 72aa4f3b gnu-toolchain/stage2-binutils.bst 
fetch needed dbb0b7ba gnu-toolchain/stage2-busybox.bst 
     waiting a4774fa2 gnu-toolchain/stage2-fake-bash.bst 
fetch needed 3e3f53d0 gnu-toolchain/stage2-gawk.bst 
fetch needed 61c71f36 gnu-toolchain/stage2-gcc-fixed-headers.bst 
fetch needed 8617fec8 gnu-toolchain/stage2-libstdcxx.bst 
fetch needed 1eb11709 gnu-toolchain/stage2-gcc.bst 
fetch needed af38df08 gnu-toolchain/stage2-make.bst 
     waiting c0ac6a25 gnu-toolchain/stage2-reset-specs.bst 
     waiting 362f602b gnu-toolchain/stage2.bst 
fetch needed 495cfcd4 gnu-toolchain/fhs-dirs.bst 
fetch needed 437eae39 gnu-toolchain/linux-api-headers.bst 
fetch needed d0efee0f gnu-toolchain/glibc.bst 
fetch needed fb9edf71 gnu-toolchain/zlib.bst 
fetch needed f3bc509b gnu-toolchain/binutils.bst 
fetch needed bb8dfc34 gnu-toolchain/busybox.bst 
fetch needed 75a98368 gnu-toolchain/ccache.bst 
fetch needed 41117f13 gnu-toolchain/gawk.bst 
fetch needed f414eabf gnu-toolchain/m4-tarball.bst 
fetch needed 75a261c6 gnu-toolchain/gcc.bst
```

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @devcurmudgeon]](https://gitlab.com/devcurmudgeon) on Apr 5, 2018, 19:50

changed the description

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 6, 2018, 10:42

Agreed, I'm not sure if the default for `bst show` should change for this to use the `%{full-key}` instead of the `%{key}` format, probably not.

I am concerned about text wrapping in the terminal here, though.

Note that right now at least, this is already [configurable in user configuration](http://buildstream.gitlab.io/buildstream/config.html), adding the following to your `~/.config/buildstream.conf` will show you the full keys at startup:

```yaml
logging:

  element-format: |

    %{state: >12} %{full-key} %{name} %{workspace-dirs}
```

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 6, 2018, 12:48

closed via commit c55dad4be4249e53f0cee56500a6c0772fc85c35

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 7, 2018, 11:05

mentioned in commit c55dad4be4249e53f0cee56500a6c0772fc85c35
